In [ ]:
# Predict who could survive in Titanic Disaster

In [40]:
import pandas as pd

In [41]:
# Reading training csv
path = './train.csv'
df = pd.read_csv(path)

In [42]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [43]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [44]:
df.size

10692

### Columns interpretation

1. Survived -> The passenger survive? (1 yes, 0 no)
2. Pclass -> Ticket class (first, second and third)
3. SibSp -> Number of **siblings or spouses aboard**
4. Parch -> Number of **parents or children aboard**
5. Embarked -> Port of embarkation


## How was the people who survived?

In [35]:
survived = df.loc[df.Survived == 1]
print(survived.size)
survived

1476


PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
10            11         1       3   
11            12         1       1   
21            22         1       2   
..           ...       ...     ...   
862          863         1       1   
871          872         1       1   
879          880         1       1   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
21                               Beesley, Mr. Lawrence    male  34.0      0   
..                                                 ...     ...   ...    ...   
862  Swift, Mrs. Frederick Joel (Margaret Welles Ba...  female  48.0      0   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch    Ticket     Fare Cabin Embarked  
1        0  PC 17599  71.2833   C85        C  
3        0    113803  53.1000  C123        S  
10       1   PP 9549  16.7000    G6        S  
11       0    113783  26.5500  C103        S  
21       0    248698  13.0000   D56        S  
..     ...       ...      ...   ...      ...  
862      0     17466  25.9292   D17        S  
871      1     11751  52.5542   D35        S  
879      1     11767  83.1583   C50        C  
887      0    112053  30.0000   B42        S  
889      0    111369  30.0000  C148        C  

[123 rows x 12 columns]

Only 38% of the people aboard, survived (4104 out of 10,692)

In [36]:
survived['Pclass'].value_counts()

1    106
2     12
3      5
Name: Pclass, dtype: int64

In [37]:
survived.loc[survived.Pclass == 1, 'Sex'].value_counts()

female    71
male      35
Name: Sex, dtype: int64

In [38]:
survived['Cabin'].value_counts()

B96 B98    4
F33        3
E121       2
D36        2
D          2
          ..
E33        1
E40        1
C32        1
E34        1
C148       1
Name: Cabin, Length: 94, dtype: int64

TODO: How to handle with nan values, dropping them out will cause a side-effect on the prediction model since it is a big amount